### PR TITLE
add Scrabble plugin

### PR DESCRIPTION
**Pull request by gooh**
allows Jeeves to calculate the Scrabble score of the passed input string. Currently only supports the original english scoring.

### CODE REVIEW

File: src/Plugins/Scrabble.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    {
+        return 'Calculates the Scrabble score for the given input';
+    }
+
+    private function calculateScore(string $input): string
+    {
+        $sum = 0;
+        foreach ($input as $char) {
+            if ($char !== '~') {
+                $sum += key(preg_grep("~$char~i", SCORES['en']));
```
Comment (kelunik) [inline src/Plugins/Scrabble.php]: You probably want to use `preg_quote` here and escape `$char`.

File: src/Plugins/Scrabble.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    public function __construct(ChatClient $chatClient)
+    {
+        $this->chatClient = $chatClient;
+    }
+
+    public function scrabble(Command $command): Promise
+    {
+        if (!$command->hasParameters()) {
+            return new Success();
```
Comment (kelunik) [inline src/Plugins/Scrabble.php]: How about posting a help message here in that case?

**Comment by gooh** [comment]
Might make more sense to hook this up against https://www.wordgamedictionary.com instead in the long run

File: src/Plugins/Scrabble.php [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    public function getDescription(): string
+    {
+        return 'Calculates the Scrabble score for the given input';
+    }
+
+    public function scrabble(Command $command): Promise
+    {
+        if ($command->hasParameters() === false) {
```
Comment (kelunik) [inline src/Plugins/Scrabble.php]: Simply use `if (!$command->hasParameters()) {`

File: src/Plugins/Scrabble.php [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+    private function formatScores(array $scores): string
+    {
+        $total = 0;
+        $result = '';
+        foreach ($scores as $word => $score) {
+            $result .= "$word = $score, ";
+            $total += $score;
+        }
+        return $result .= "TOTAL SCORE = $total";
```
Comment (kelunik) [inline src/Plugins/Scrabble.php]: Do we really need CAPS here? How about `word1 (score1) + word2 (score2) makes XY in total.`? Jeeves shouldn't sound like a robot. :stuck_out_tongue_winking_eye:

File: src/Plugins/Scrabble.php (hunk 3 above)
Comment (gooh) [inline src/Plugins/Scrabble.php]: I don't think it matters whether it's the short form or explicit. We have both in the code base, so it doesn't need fixing.

File: src/Plugins/Scrabble.php (hunk 4 above)
Comment (gooh) [inline src/Plugins/Scrabble.php]: Feel free to amend. I think it's good enough as is.

Comment (kelunik) [inline src/Plugins/Scrabble.php]: Yes it is. This was meant as a single comment, but accidentally ended up in the review.

File: src/Plugins/Scrabble.php (hunk 3 above)
Comment (gooh) [inline src/Plugins/Scrabble.php]: Aguideline for this is pretty overzealous imo. Also, consider that `!expr` is not equivalent to `expr === false` due to type checking, e.g. `[] === false` is not the same as `![]`. It doesn't matter in this specific piece of code, but you need to be able to use both (`strpos`, …).

**Pull request merged**